### PR TITLE
Release v2.3.0.1

### PR DESCRIPTION
**Pull request by IshaanDesai**
Release for [preCICE v2.3.0](https://github.com/precice/precice/releases/tag/v2.3.0)

Main features:
* New API function to check if mesh connectivity is required https://github.com/precice/matlab-bindings/pull/31
* New **experimental** API functions for direct mesh access https://github.com/precice/matlab-bindings/pull/32

### CODE REVIEW

File: +precice/@SolverInterface/private/preciceGateway.cpp [hunk 1]
```diff
@@ -45,6 +45,9 @@ enum class FunctionID {
     setMeshTriangleWithEdges = 51,
     setMeshQuad = 52,
     setMeshQuadWithEdges = 53,
+    setMeshAccessRegion = 55,
+    getMeshVerticesAndIDs = 56,
+    isMeshConnectivityRequired = 54,
```
Comment (BenjaminRodenberg) [inline +precice/@SolverInterface/private/preciceGateway.cpp:50]: Why not `54, 55, 56`? (Probably does not matter)

File: +precice/@SolverInterface/private/preciceGateway.cpp [hunk 2]
```diff
@@ -313,7 +316,38 @@ class MexFunction: public matlab::mex::Function {
                 interface->setMeshQuadWithEdges(meshID[0],firstVertexID[0],secondVertexID[0],thirdVertexID[0],fourthVertexID[0]);
                 break;
             }
+<<<<<<< HEAD
```
Comment (BenjaminRodenberg) [inline +precice/@SolverInterface/private/preciceGateway.cpp:319]: This is a leftover from an incomplete merge.

**Review by BenjaminRodenberg — CHANGES_REQUESTED** [review]
Could you also run a quick test with the new API functions? No need to do something real, but just calling the functions to ensure they work would be good.

File: +precice/@SolverInterface/private/preciceGateway.cpp (hunk 1 above)
Comment (IshaanDesai) [reply +precice/@SolverInterface/private/preciceGateway.cpp:50]: Mixing up during merging both pull requests is definitely possible. In fact the order is not even incorrect, just the function name `isMeshConnectivityRequired` moved below the direct mesh access functions somehow. I will correct this.

File: +precice/@SolverInterface/SolverInterface.m [hunk 3]
```diff
@@ -193,6 +193,22 @@ function setMeshQuadWithEdges(obj, meshID, firstVertexID, secondVertexID, thirdV
...
+            preciceGateway(uint8(55),int32(meshID),boundingBox)
+        end
+
+        % getMeshVerticesAndIDs  - EXPERIMENTAL
+        function [vertices, outIDs] = getMeshVerticesAndIDs(meshID)
+            inSize = getMeshVertexSize(meshID);
+            [vertices,outIDs] = preciceGateway(uint8(56),int32(meshID),int32(inSize));
+        end
+
+        % isMeshConnectivityRequired
```
Comment (BenjaminRodenberg) [inline +precice/@SolverInterface/SolverInterface.m:207]:
```suggestion
        % isMeshConnectivityRequired - EXPERIMENTAL
```
?

Or where does this function come from?

File: +precice/@SolverInterface/SolverInterface.m [hunk 4]
```diff
@@ -193,6 +193,22 @@ function setMeshQuadWithEdges(obj, meshID, firstVertexID, secondVertexID, thirdV
             preciceGateway(uint8(53),int32(meshID),int32(firstVertexID),int32(secondVertexID),int32(thirdVertexID),int32(fourthVertexID));
         end
         
+        % setMeshAccessRegion - EXPERIMENTAL
+        function setMeshAccessRegion(meshID, boundingBox)
+            preciceGateway(uint8(55),int32(meshID),boundingBox)
+        end
+
+        % getMeshVerticesAndIDs  - EXPERIMENTAL
```
Comment (BenjaminRodenberg) [inline +precice/@SolverInterface/SolverInterface.m:201]:
```suggestion
        % getMeshVerticesAndIDs - EXPERIMENTAL
```
Just for consistency.

File: +precice/@SolverInterface/private/preciceGateway.cpp [hunk 5]
```diff
@@ -313,6 +316,34 @@ class MexFunction: public matlab::mex::Function {
...
+                buffer_ptr_t<int32_t> ids_ptr = factory.createBuffer<int32_t>(size[0]);
+                int32_t* ids = ids_ptr.get();
+                buffer_ptr_t<double> positions_ptr = factory.createBuffer<double>(size[0]);
+                double* positions = positions_ptr.get();
+                interface->getMeshVertices(meshID[0],size[0],ids,positions);
+                outputs[0] = factory.createArrayFromBuffer<double>({1,size[0]}, std::move(positions_ptr));
+                outputs[1] = factory.createArrayFromBuffer<int32_t>({1,size[0]}, std::move(ids));
+                break;
+            } 
+
```
Comment (BenjaminRodenberg) [inline +precice/@SolverInterface/private/preciceGateway.cpp:339]:
```suggestion
```
Again: Consistency, but I think there are also a few places where this is different, below. You can also ignore this. As long as we don't have a stylechecker it does not make a lot of sense investing effort here (if it is just about blank lines and spaces).

**Review by BenjaminRodenberg — APPROVED** [review]
Only minor points. Ready to go from my perspective :tada:

**Pull request merged**